File: databinding/src/main/java/com/example/anvil/databinding/RadioGroupLayout.java
package com.example.anvil.databinding;

import android.content.Context;
import android.widget.LinearLayout;
import android.widget.RadioGroup;

import static trikita.anvil.DSL.*;
import trikita.anvil.Anvil;
import trikita.anvil.RenderableView;

public class RadioGroupLayout extends RenderableView {

    int index;

    public RadioGroupLayout(Context c) {
        super(c);
    }

    public void view() {
        linearLayout(() -> {
            size(FILL, FILL);
            orientation(LinearLayout.VERTICAL);

            radioGroupLayout(100);
            radioGroupLayout(200);
        });
    }

    private void radioGroupLayout(final int startId) {
        radioGroup(() -> {
            margin(dip(20));
            onCheckedChange((RadioGroup rg, int id) -> {
                index = id - startId;
            });
            for (int i = 0; i < 3; i++) {
                final int index = i;
                radioButton(() -> {
                    id(startId + index);
                });
            }
            //final RadioGroup r = Anvil.currentView();
            //r.check(startId + index);
            check(startId + index);
        });
    }
}

File: databinding-kotlin/src/main/kotlin/com/example/anvil/databinding/RadioGroupLayout.kt
package com.example.anvil.databinding

import android.content.Context
import android.widget.LinearLayout
import android.widget.RadioGroup

import trikita.anvil.DSL.*
import trikita.anvil.Anvil
import trikita.anvil.RenderableView

class RadioGroupLayout(c: Context) : RenderableView(c) {

    var index: Int = 0

    override fun view() {
        linearLayout {
            size(FILL, FILL)
            orientation(LinearLayout.VERTICAL)

            radioGroupLayout(100)
            radioGroupLayout(200)
        }
    }

    private fun radioGroupLayout(startId: Int) {
        radioGroup {
            margin(dip(20))
            for (i in 0..2) {
                val index = i
                radioButton { id(startId + index) }
            }
            check(startId + index)
            onCheckedChange { rg: RadioGroup, id: Int -> index = id - startId }
        }
    }
}

